### PR TITLE
Introduces compact diff view. Closes #1831

### LINKED ISSUE #1831: Make it easier to understand what the application diffs are
**Is your feature request related to a problem? Please describe.**

A nice feature of `kapp` is that it prints what it's about to do before it does it. It'd be nice to see what sync is going to do before 

**Describe the solution you'd like**

Display UI screen showing the plan before it is applied.

### PR DESCRIPTION
**Pull request by alexec**
### Modified (Compact View)
compact

### Modified (Full View)
modified
### Added
added

### Deleted 
deleted

### Side-By-Side
side-by-side

### CODE REVIEW

**Comment by alexec** [comment]
@alexmt let me know if you can figure this out using Monaco

File: ui/src/app/applications/components/application-resource-diff/application-resource-diff.tsx [hunk 1]
```diff
@@ -16,42 +17,61 @@ export interface ApplicationComponentDiffProps {
...
-                    </label>
+            const a = live ? jsYaml.safeDump(live, {indent: 2}) : '';
+            const b = target ? jsYaml.safeDump(target, {indent: 2}) : '';
+            const context = pref.appDetails.compactDiff ? 2 : Number.MAX_SAFE_INTEGER;
+            const diffText = formatLines(diffLines(a, b), {context});
+            const files = parseDiff(diffText, {nearbySequences: 'zip'});
+            const viewType = pref.appDetails.inlineDiff ? 'unified' : 'split';
+            return (
+                <div className='application-component-diff'>
+                    <div className='application-component-diff__checkboxes'>
```
Comment (alexmt) [inline ui/src/app/applications/components/application-resource-diff/application-resource-diff.tsx]: Please don't forget to add left/right padding to have some extra space between checkboxes

**Comment by alexmt** [comment]
I feel like compact diff provides more benefits than syntax highlighting in diff view. I'm to stop using Monaco, loose syntax highlighting in diff view and create ticket to try to get it back.

@jessesuen , what do you think?

**Comment by alexec** [comment]
build failure un-related

File: ui/src/app/applications/components/application-resources-diff/application-resources-diff.scss [hunk 2]
```diff
@@ -0,0 +1,14 @@
+.application-component-diff {
```
Comment (alexmt) [inline ui/src/app/applications/components/application-resources-diff/application-resources-diff.scss]: please rename `application-component-diff` to `application-resources-diff`

**Review by alexmt — CHANGES_REQUESTED** [review]
Added one minor comment. Also I would like to request one enhancement: 

I think resources on full app diff tab should sorted by sync status, then by group/kind/name. In this case resources without diffs should go last. Application might include a lot of resources. E.g. HA Argo CD includes 34 top level resources. It would be much better show resource with differences first, so user don't have to scroll a lot

image

File: ui/src/app/applications/components/application-resources-diff/application-resources-diff.scss (hunk 2 above)
Comment (alexec) [reply ui/src/app/applications/components/application-resources-diff/application-resources-diff.scss]: I've hidden unmodified files.

**Review by alexmt — APPROVED** [review]

**Pull request merged**